src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -143,7 +143,11 @@ public virtual void CleanUp()
             }
 
             // Add Current package version to package versions list.
-            _allPackageVersions = new List<NuGetVersion>() { searchResultPackage.Version };
+            _allPackageVersions = new List<(NuGetVersion version, bool isDeprecated)>()
+            {
+                (searchResultPackage.Version, (await searchResultPackage.GetPackageDeprecationMetadataAsync()) != null)
```

Comment:
We've already discussed this as too much to refactor, but the lack of separation between UI and business logic code makes me sad. Particularly since I've been working on some UI delay issues.

Can `searchResultPackage.GetPackageDeprecationMetadataAsync()` make an HTTP call, or is the data already in memory, and it's just searching data structures? Does this code happen on the UI thread, or a background thread?

Reply:
I completely agree. I think this code will need a large refactor eventually or else the situation will continue to get progressively worse, as it was fundamentally written in a way that did not separate the UI layer from the business logic.

I looked into this more. It is wrong to fetch the deprecation information at this point--the idea is that the code inserts the current version with nothing else to bypass the case in which fetching the versions (or, in this case, the deprecation metadata) takes a long time. I should insert "false" here instead of making a call.